src/render/painter.js

```diff
@@ -497,14 +508,46 @@ class Painter {
             }
         }
 
+        // Terrain depth render ===========================================
```

Comment:
Moving this before [rebinding default framebuffer in L460](https://github.com/mapbox/mapbox-gl-js/pull/10258/files#diff-687df4b4895a38c16c3509d0ca0069ebdfa5b351ab55230b1ed1b33126b7908cL460) would  save on GPU load store - avoid the need to rebind again in https://github.com/mapbox/mapbox-gl-js/blob/15f0983b29ebcc146d0e2ccf37fb78ea941caf00/src/terrain/terrain.js#L1012 

That would require also moving some of the initialization up (painter.depthRangeFor3D).

Reply:
Addressed in https://github.com/mapbox/mapbox-gl-js/pull/10258/commits/253d8233000cb0e6aa124c14b79ecc6243ed7c13